### PR TITLE
DP-32007 [a11y] Download link normal state indicator

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

- Add a 1-px underline to indicate a link object to a download link at normal state matching the RTE link style
Screenshot 2024-04-19 at 11 09 31 AM

- Change the link underline color from Bay Blue 50% tint to Bay Blue 40 % tint to meet the WCAG required color contrast 3:1 against the white background
- Set up a new color variable for the link underline as `$c-link` (cf, link text has its own variable as `$c-font-link`)
    -  The color is applied to non-text elements to indicate a link (borders, underlines, icons) 
    - The color is used for a normal and a hover state of links
    - The variables used for link underlines (`$c-primary-lighter`, `$c-bay-blue-lighter`, and the variations of rgba settings [e,g, `rgba($c-primary, .5)`] ) are consolidated to `$c-link`
    - `$c-primary-lighter` and `$c-bay-blue-lighter` remain to be Bay Blue **50% tint**  (any decorative elements with these variables retain their intended colors. )
Screenshot_2024-04-19_at_11_23_07 AM

- Add an underline at hover to the button links for the show/hide (expand/close) to indicate the interactive element
     - The same visual components in different templates had inconsistent behaviors (show or not show an underline at hover).  
     - Set `$c-link` for their chevron on the right and underline at hover
**Before :**
Screenshot 2024-04-19 at 11 35 44 AM

**After:**
Screenshot 2024-04-19 at 11 36 01 AM

-  [TEST] Change the color of the arrow icon with links to match the link text color to meet the WCAG required color contrast against the light blue background of the card and the callout link 
     - Use the variable `$c-font-link` as `fill: $c-font-link` (if we decide to the change color or set up a color variable for the arrow icon, they can be found easily) 
Screenshot 2024-04-19 at 11 16 54 AM

**Note:  The Backstop reference screenshots haven't been updated yet.**

## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-32007)
- [Github issue]()

## Steps to Test

1.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: packages/assets/scss/02-molecules/_download-link.scss [hunk 1]
```diff
@@ -65,7 +65,7 @@
   &__file-link {
 
     @include ma-link-underline;
-    padding-bottom: 3px;
+    padding-bottom: 1px;//3px;
```
Comment (clairesunstudio) [inline packages/assets/scss/02-molecules/_download-link.scss]:
```suggestion
    padding-bottom: 1px;

```

File: packages/assets/scss/02-molecules/_listing-table.scss [hunk 2]
```diff
@@ -125,7 +125,7 @@
       font-size: $fonts-small;
 
       &:after {
-        opacity: .5;
+        opacity: .4;//.5;  Match the opacity to $c-link.
```
Comment (clairesunstudio) [inline packages/assets/scss/02-molecules/_listing-table.scss]:
```suggestion
        opacity: .4; // Match the opacity to $c-link.

```

File: packages/assets/scss/03-organisms/_teaser.scss [hunk 3]
```diff
@@ -141,7 +141,7 @@
       }
 
       svg {
-        fill: rgba($c-primary, 0.5);
+        fill: $c-font-link;// rgba($c-primary, 0.5);
```
Comment (clairesunstudio) [inline packages/assets/scss/03-organisms/_teaser.scss]:
```suggestion
        fill: $c-font-link;
```

File: packages/assets/scss/03-organisms/_sticky-toc.scss [hunk 4]
```diff
@@ -278,19 +278,24 @@
...
       &::after {
         content: "";
         position: absolute;
         top: 50%;
         transform: translateY(-50%) rotate(45deg);
         right: 0;
         display: inline-block;
-        border-right: 4px solid rgba($c-primary, 0.3);
-        border-bottom: 4px solid rgba($c-primary, 0.3);
+        border-right: 4px solid $c-font-link;//rgba($c-primary, 0.3);
```
Comment (clairesunstudio) [inline packages/assets/scss/03-organisms/_sticky-toc.scss]:
```suggestion
        border-right: 4px solid $c-font-link;
```

File: packages/assets/scss/03-organisms/_sticky-toc.scss [hunk 5]
```diff
@@ -278,19 +278,24 @@
...
+        border-bottom: 4px solid $c-font-link;//rgba($c-primary, 0.3);
```
Comment (clairesunstudio) [inline packages/assets/scss/03-organisms/_sticky-toc.scss]:
```suggestion
        border-bottom: 4px solid $c-font-link;
```

File: packages/assets/scss/03-organisms/_related-orgs-and-topics.scss [hunk 6]
```diff
@@ -108,8 +113,8 @@
       transform: translateY(-50%) rotate(45deg);
       right: 0;
       display: inline-block;
-      border-right: 4px solid rgba($c-primary, 0.3);
-      border-bottom: 4px solid rgba($c-primary, 0.3);
+      border-right: 4px solid $c-link; //rgba($c-primary, 0.3);
```
Comment (clairesunstudio) [inline packages/assets/scss/03-organisms/_related-orgs-and-topics.scss]:
```suggestion
      border-right: 4px solid $c-link;
```

File: packages/assets/scss/03-organisms/_related-orgs-and-topics.scss [hunk 7]
```diff
@@ -108,8 +113,8 @@
...
+      border-bottom: 4px solid $c-link; //rgba($c-primary, 0.3);
```
Comment (clairesunstudio) [inline packages/assets/scss/03-organisms/_related-orgs-and-topics.scss]:
```suggestion
      border-bottom: 4px solid $c-link;
```

**Comment by clairesunstudio** [comment]
The above suggestions are all about cleaning up the code comments. 

I reviewed the Mayflower backstop report, all looks good except for the file icons colors are lighter now. I think we should at least set that to $c-link or even $c-font-link. What do you think?
Screen Shot 2024-05-14 at 10 46 36 AM
Screen Shot 2024-05-14 at 10 46 39 AM

I also noticed a change in the footnote size. Can you please confirm that it's expected? 
Screen Shot 2024-05-14 at 12 04 49 PM
Screen Shot 2024-05-14 at 12 04 53 PM

**Comment by ygannett** [comment]
The [backstop failure](https://output.circle-artifacts.com/output/job/a97e5723-ceb7-4c4e-8d4a-ba5646166b07/artifacts/0/backstop/reports/html/index.html) is not related to this PR's change.

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**